qcodes/instrument/meta/meta_instrument.py

```diff
@@ -0,0 +1,185 @@
+
+    def __init__(
+        self,
+        name: str,
+        station: Station,
+        aliases: Dict[str, List[str]],
+        initial_values: Dict[str, Any] = None,
+        set_initial_values_on_load: bool = False,
+        setters: Dict[str, Dict[str, Any]] = None,
+        units: Dict[str, Dict[str, str]] = None,
+        metadata: Optional[Dict[Any, Any]] = None):
+        super().__init__(name=name, metadata=metadata)
+        self._add_parameters(
+            station=station,
+            aliases=aliases,
+            setters=setters or {},
+            units=units or {}
+        )
+        self._initial_values = initial_values or {}
+        if set_initial_values_on_load:
+            self.set_initial_values()
+
+    def set_initial_values(self, dry_run: bool = False):
+        """Set parameter initial values on meta instrument
+
+        Args:
+            dry_run: Dry run to test if defaults are set correctly.
+                Defaults to False.
+        """
+        _log.debug(f"Setting default values: {self._initial_values}")
+        for path, value in self._initial_values.items():
+            param = self.parse_instrument_path(self, path=path)
+            msg = f"Setting parameter {self.name}.{path} to {value}."
+            if not dry_run:
+                _log.debug(msg)
+                if hasattr(param, "set"):
+                    param.set(value)
+                else:
+                    _log.debug("No set method found, trying to assign value.")
+                    if "." in path:
+                        name = path.split(".")[-1]
+                        parent_path = ".".join(path.split(".")[:-1])
+                        parent = self.parse_instrument_path(self, path=parent_path)
+                    else:
+                        parent, name = self, path
+                    print(parent, name, value)
+                    setattr(parent, name, value)
+            else:
+                print(f"Dry run: {msg}")
+
+    def _add_parameters(
+        self,
+        station: Station,
+        aliases: Dict[str, List[str]],
+        setters: Dict[str, Dict[str, Any]],
+        units: Dict[str, Dict[str, str]]
+    ):
+        """Add parameters to meta instrument based on specified aliases, endpoints
+        and setter methods"""
+        for param_name, paths in aliases.items():
+            self._add_parameter(
+                param_name=param_name,
+                station=station,
+                paths=paths,
+                setter=setters.get(param_name),
+                unit=units.get(param_name)
+            )
+
+    @staticmethod
+    def _endpoint_names(endpoints: List[Parameter]):
+        """Get the endpoint names"""
+        endpoint_names = [_e.name for _e in endpoints]
+        if len(endpoint_names) != len(set(endpoint_names)):
+            endpoint_names = [
+                f"{_e}{n}" for n, _e in enumerate(endpoint_names)
+            ]
+        return endpoint_names
+
+    def _add_parameter(
```

Comment:
perhaps `_create_and_add_parameter` to clarify what more than add_parameter it does